File: kotlin-node/karakum/src/jsMain/resources/plugins/convertUtilTypeHelpers.js
import ts from "typescript";

export default function (node, context, render) {
    const sourceFileName = node.getSourceFile()?.fileName ?? "generated.d.ts"

    if (
        sourceFileName.endsWith("util.d.ts")
        && ts.isTypeReferenceNode(node)
        && ts.isIdentifier(node.typeName)
        && (
            node.typeName.text === "IfDefaultsTrue"
            || node.typeName.text === "IfDefaultsFalse"
        )
    ) {
        return ""
    }


    return null
}

File: kotlin-node/karakum/src/jsMain/kotlin/node/karakum/plugins/convertUtilTypeHelpers.kt
package node.karakum.plugins

import node.karakum.util.nullable
import io.github.sgrishchenko.karakum.extension.createPlugin
import io.github.sgrishchenko.karakum.util.getSourceFileOrNull
import typescript.isIdentifier
import typescript.isTypeReferenceNode

val convertUtilTypeHelpers = createPlugin { node, context, render ->
    nullable {
        val sourceFileName = ensureNotNull(node.getSourceFileOrNull()).fileName
        ensure(sourceFileName.endsWith("util.d.ts"))

        ensure(isTypeReferenceNode(node))

        val typeName = node.typeName
        ensure(isIdentifier(typeName))
        ensure(typeName.text == "IfDefaultsTrue" || typeName.text == "IfDefaultsFalse")

        ""
    }
}
